### DISCUSSION TITLE
function export abbriviation clearup

### DISCUSSION THREAD

**Discussion by kam1986** [WebAssembly/spec#1704] [Q&A]
on page 117 in the documentation v2.0 the formulations

**Abbreviations**
Multiple anonymous locals may be combined into a single declaration:
```
‘(’ ‘local’ valtype* ‘)’ ≡ (‘(’ ‘local’ valtype ‘)’)*
```
Functions can be defined as _imports_ or _exports_ inline:
```
‘(’ ‘func’ id?
‘(’ ‘import’ name1 name2 ‘)’ typeuse ‘)’ ≡
‘(’ ‘import’ name1 name2 ‘(’ ‘func’ id? typeuse ‘)’ ‘)’
‘(’ ‘func’ id? ‘(’ ‘export’ name ‘)’ . . . ‘)’ ≡
‘(’ ‘export’ name ‘(’ ‘func’ id′ ‘)’ ‘)’ ‘(’ ‘func’ id′ . . . ‘)’
(if id? ̸= 𝜖 ∧ id′ = id? ∨ id? = 𝜖 ∧ id′ fresh)
```
**Note:** The latter abbreviation can be applied repeatedly, if “. . .” contains additional export clauses. Consequently,
a function declaration can contain any number of exports, possibly followed by an import.

I'm a little confused about the 'note'. is it correctly understood that it simply means that we can export a function with multiple export names?

**Comment by rossberg** [comment]
Yes, and that imported functions can be exported as well.
